### PR TITLE
ceph-objectstore-tool: don't destroy SnapMapper until the txn is completed

### PR DESCRIPTION
**Pull request by tchaikov**
and some cleanups

### CODE REVIEW

**Comment by tchaikov** [comment]
this PR contains #20589

File: src/tools/ceph_objectstore_tool.cc [hunk 1]
```diff
@@ -415,6 +413,22 @@ int mark_pg_for_removal(ObjectStore *fs, spg_t pgid, ObjectStore::Transaction *t
 #pragma GCC diagnostic pop
 #pragma GCC diagnostic warning "-Wpragmas"
 
+template<typename Func>
+void wait_until_done(ObjectStore::Transaction* txn, Func&& func)
```
Comment (dzafman) [inline src/tools/ceph_objectstore_tool.cc]: Why is this mechanism needed now?

Comment (tchaikov) [reply src/tools/ceph_objectstore_tool.cc]: see the commit message of https://github.com/ceph/ceph/pull/20593/commits/42f310f253a658ec8c600a8406084ec91ecd49ba

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2018-02-27_10:33:49-rados-wip-kefu-testing-2018-02-27-1348-distro-basic-mira/

**Pull request merged**